### PR TITLE
feat: Add error messages to providers that don't support query options

### PR DESCRIPTION
**Pull request by DRK3**
- These new error messages will help ensure that someone won't get unexpected results when using those options in a Query on a storage provider that doesn't support them.
- The unit tests have been updated to avoid calling the new common storage Query tests that use those unsupported options.

Signed-off-by: Derek Trider <Derek.Trider@securekey.com>

### CODE REVIEW

**Review by bstasyszyn — APPROVED** [review]

**Review by aholovko — APPROVED** [review]

File: cmd/aries-agent-mobile/go.mod [hunk 1]
```diff
@@ -8,10 +8,10 @@ go 1.16
...
 require (
 	github.com/google/uuid v1.1.2
-	github.com/hyperledger/aries-framework-go v0.1.7-0.20210409151411-eeeb8508bd87
-	github.com/hyperledger/aries-framework-go/component/storageutil v0.0.0-20210409151411-eeeb8508bd87
-	github.com/hyperledger/aries-framework-go/spi v0.0.0-20210421165342-de8f911415e3
-	github.com/hyperledger/aries-framework-go/test/component v0.0.0-20210409151411-eeeb8508bd87
+	github.com/hyperledger/aries-framework-go v0.1.7-0.20210422133815-2ef2d99cb692
+	github.com/hyperledger/aries-framework-go/component/storageutil v0.0.0-20210422133815-2ef2d99cb692
+	github.com/hyperledger/aries-framework-go/spi v0.0.0-20210422133815-2ef2d99cb692
+	github.com/hyperledger/aries-framework-go/test/component v0.0.0-20210422133815-2ef2d99cb692
```
Comment (Baha-sk) [inline cmd/aries-agent-mobile/go.mod:14]: missing equivalent go.sum update for this

File: cmd/aries-agent-rest/go.mod [hunk 2]
```diff
@@ -10,10 +10,10 @@ require (
...
 	github.com/golang/snappy v0.0.3 // indirect
 	github.com/gorilla/mux v1.7.3
-	github.com/hyperledger/aries-framework-go v0.1.7-0.20210409151411-eeeb8508bd87
-	github.com/hyperledger/aries-framework-go/component/storage/leveldb v0.0.0-20210421205521-3974f6708723
-	github.com/hyperledger/aries-framework-go/component/storageutil v0.0.0-20210409151411-eeeb8508bd87
-	github.com/hyperledger/aries-framework-go/spi v0.0.0-20210421205521-3974f6708723
+	github.com/hyperledger/aries-framework-go v0.1.7-0.20210422133815-2ef2d99cb692
+	github.com/hyperledger/aries-framework-go/component/storage/leveldb v0.0.0-20210422133815-2ef2d99cb692
+	github.com/hyperledger/aries-framework-go/component/storageutil v0.0.0-20210422133815-2ef2d99cb692
+	github.com/hyperledger/aries-framework-go/spi v0.0.0-20210422133815-2ef2d99cb692
```
Comment (Baha-sk) [inline cmd/aries-agent-rest/go.mod:16]: same here

File: cmd/aries-js-worker/go.mod [hunk 3]
```diff
@@ -8,10 +8,9 @@ go 1.16
 
 require (
 	github.com/google/uuid v1.1.2
-	github.com/hyperledger/aries-framework-go v0.1.7-0.20210409151411-eeeb8508bd87
+	github.com/hyperledger/aries-framework-go v0.1.7-0.20210422133815-2ef2d99cb692
```
Comment (Baha-sk) [inline cmd/aries-js-worker/go.mod:11]: and here

File: test/bdd/go.mod [hunk 4]
```diff
@@ -20,10 +20,10 @@ require (
...
 	github.com/google/uuid v1.1.2
 	github.com/gorilla/mux v1.8.0
-	github.com/hyperledger/aries-framework-go v0.1.6-0.20210409151411-eeeb8508bd87
-	github.com/hyperledger/aries-framework-go/component/storage/leveldb v0.0.0-20210421205521-3974f6708723
-	github.com/hyperledger/aries-framework-go/component/storageutil v0.0.0-20210409151411-eeeb8508bd87
-	github.com/hyperledger/aries-framework-go/spi v0.0.0-20210421205521-3974f6708723
+	github.com/hyperledger/aries-framework-go v0.1.7-0.20210422133815-2ef2d99cb692
+	github.com/hyperledger/aries-framework-go/component/storage/leveldb v0.0.0-20210422133815-2ef2d99cb692
+	github.com/hyperledger/aries-framework-go/component/storageutil v0.0.0-20210422133815-2ef2d99cb692
+	github.com/hyperledger/aries-framework-go/spi v0.0.0-20210422133815-2ef2d99cb692
```
Comment (Baha-sk) [inline test/bdd/go.mod:26]: also here

File: cmd/aries-agent-rest/go.mod (hunk 2 above)
Comment (DRK3) [reply cmd/aries-agent-rest/go.mod:16]: Actually, these imports aren't in go.sum to begin with. I assume this is due to the replace.

File: cmd/aries-js-worker/go.mod (hunk 3 above)
Comment (DRK3) [reply cmd/aries-js-worker/go.mod:11]: Actually, these imports aren't in go.sum to begin with. I assume this is due to the replace.

File: test/bdd/go.mod (hunk 4 above)
Comment (DRK3) [reply test/bdd/go.mod:26]: Actually, these imports aren't in go.sum to begin with. I assume this is due to the replace.

**Review by fqutishat — APPROVED** [review]

File: cmd/aries-agent-mobile/go.mod (hunk 1 above)
Comment (Baha-sk) [reply cmd/aries-agent-mobile/go.mod:14]: we should probably use v0 in this case.. it's ok for now, it can be done later

**Review by Baha-sk — APPROVED** [review]

**Pull request merged**